sdktests/server_side_persistence_base.go

```diff
@@ -673,9 +673,12 @@ func basicFlagValidationMatcher(key string, version int, value string) m.Matcher
 	)
 }
 
-func basicDeletedFlagValidationMatcher(version int) m.Matcher {
+func basicDeletedFlagValidationMatcher(key string, version int) m.Matcher {
 	return m.AllOf(
-		m.JSONProperty("key").Should(m.Equal("$deleted")),
+		m.AnyOf(
+			m.JSONProperty("key").Should(m.Equal(key)),
```

Comment:
I'm throwing this up here because it was quick to implement, but we should stop and decide if we actually want this.

Should we update each integration to require that the key, when deleted, is updated to $deleted?